deeprank2/query.py

```diff
@@ -93,28 +95,45 @@ def _check_pssm(pdb_path: str, pssm_paths: Dict[str, str], suppress: bool, verbo
         _log.warning(error_message)
 
 
+# TODO: consider whether we want to use the built-in repr and eq, or define it ourselves
+# if built-in: consider which arguments to include in either.
+@dataclass(repr=False, kw_only=True)
 class DeepRankQuery:
+    """Represents one entity of interest, like a single residue variant or a protein-protein interface.
 
-    def __init__(self, model_id: str, targets: Optional[Dict[str, Union[float, int]]] = None, suppress_pssm_errors: bool = False):
-        """Represents one entity of interest, like a single-residue variant or a protein-protein interface.
+    :class:`DeepRankQuery` objects are used to generate graphs from structures, and they should be created before any model is loaded.
+    They can have target values associated with them, which will be stored with the resulting graph.
 
-        :class:`DeepRankQuery` objects are used to generate graphs from structures, and they should be created before any model is loaded.
-        They can have target values associated with them, which will be stored with the resulting graph.
-
-        Args:
-            model_id (str): The ID of the model to load, usually a .PDB accession code.
-            targets (Optional[Dict[str, Union[float, int]]], optional): Target values associated with the query. Defaults to None.
-            suppress_pssm_errors (bool, optional): Suppress error raised if .pssm files do not match .pdb files and throw warning instead.
-                Defaults to False.
-        """
-
-        self._model_id = model_id
-        self._suppress = suppress_pssm_errors
+    Args:
+        model_id (str): The ID of the model to load, usually a .PDB accession code.
+        targets (Optional[Dict[str, Union[float, int]]], optional): Target values associated with the query. Defaults to None.
+        suppress_pssm_errors (bool, optional): Suppress error raised if .pssm files do not match .pdb files and throw warning instead.
+            Defaults to False.
+    """
 
-        if targets is None:
-            self._targets = {}
-        else:
-            self._targets = targets
+    pdb_path: str
+    resolution: str
+    chain_ids: List[str] | str
+    pssm_paths: Dict[str, str] = field(default_factory=dict)
```

Comment:
Just briefly, what's the need for using `field` here and why it's not used for `chain_ids` as well?